ops/model.py

```diff
@@ -921,8 +928,48 @@ def _load(self) -> '_RelationDataContent_Raw':
             # Dead relations tell no tales (and have no data).
             return {}
 
-    def _is_mutable(self):
+    def _is_readable(self):
+        """Return if the data content can be read."""
+        if not self._hook_is_running:
+            return True
+
+        # Only remote units (and the leader unit) can read *this* app databag.
+
+        # is this an app databag?
+        if not self._is_app:
+            # all unit databags are publicly readable
+            return True
+
+        # Am I leader?
+        if self._backend.is_leader():
+            # leaders have no read restrictions
+            return True
+
+        # type guard; we should not be accessing relation data
+        # if the remote app does not exist.
+        app = self.relation.app
+        assert app is not None
+
+        # is this a peer relation?
+        if app.name == self._entity.name:
+            # per relation data is always publicly readable
+            return True
+
+        # if we're here it means: this is not a peer relation,
+        # this is an app databag, and we don't have leadership.
+
+        # is this a LOCAL app databag?
+        if self._backend.app_name == self._entity.name:
+            # minions can't read local app databags
+            return False
+
+        return True
```

Comment:
This feels like it should be more like a `_validate_read` like your already existing `_validate_write`.  It could directly raise the exceptions based on *why* the read permission doesn't exist rather than providing a more generic raise later.  Also, we need to make sure we have high confidence that this doesn't return False (or raise) in any cases where the `relation-get` hook tool would otherwise succeed.